lib/Default/MySqlDatabase.class.php

```diff
@@ -43,7 +34,7 @@ public function __construct($dbName) {
 		}
 
 		// Do we need to switch databases (e.g. for compatability db access)?
-		if (self::$selectedDbName != $dbName) {
+		if ($dbName != null &&self::$selectedDbName != $dbName) {
```

Comment:
This doesn't work in the case where `self::$dbConn` was previously opened for a "history" database, because if `$dbName === null`, then it won't switch back to the `smr_live` database. I'm not exactly sure how to fix this without knowing the name of the `smr_live` database a priori, but it also depends on how we want to handle "history" databases in general.

Reply:
I was thinking about this, too, and I looked at the other implementations of this database class, and they don't seem to be referenced anywhere in the code, so it sorta felt like to me that it did not matter for them (I was going to delete them actually). I think we can come up with something better in the future with some IOC techniques I plan to apply.

Unless I'm missing something?

Reply:
We never refer to these classes explicitly. Here's an example of how it's used:
https://github.com/smrealms/smr/blob/7a4bd93dbb42ef081f098cdccfe2fc56a7026643/engine/Default/game_play.php#L150-L152
I think hard-coding the database name will help, but that alone won't fix the "switching back" issue.